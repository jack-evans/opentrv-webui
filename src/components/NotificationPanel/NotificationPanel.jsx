import React from 'react'
import { ToastNotification } from 'carbon-components-react'

const NotificationPanel = ({ notifications }) => {
  let content
  if (notifications.length > 0) {
    content = notifications.map((notification, index) => {
      return (
        <ToastNotification
          key={index}
          title={notification.title}
          subtitle={notification.subtitle}
          caption={'Time stamp: ' + notification.timeStamp}
          kind={notification.kind || 'error'}
        />
      )
    })
  }

  return (
    <div className='NotificationPanel'>
      {content}
    </div>
  )
}

export default NotificationPanel
