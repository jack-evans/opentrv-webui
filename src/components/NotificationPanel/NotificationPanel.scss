$notification-panel-height: calc(100vh - 60px);
$notification-panel-width: 300px;

.NotificationPanel {
  background-color: transparent;
  float: right;
  height: $notification-panel-height;
  position: fixed;
  right: 0px;
  top: 60px;
  width: $notification-panel-width;
  z-index: 100000;
}