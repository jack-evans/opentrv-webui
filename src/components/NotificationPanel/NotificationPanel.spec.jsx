import React from 'react'
import NotificationPanel from './NotificationPanel'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('NotificationPanel.jsx', () => {
  describe('when notifications is an empty array', () => {
    it('renders without crashing', () => {
      expect(shallow(<NotificationPanel notifications={[]} />)).toMatchSnapshot()
    })

    it('renders 0 ToastNotifications', () => {
      const wrapper = mount(<NotificationPanel notifications={[]} />)
      expect(wrapper.find('.bx--toast-notification')).toHaveLength(0)
    })
  })

  describe('when notifications is a populated array', () => {
    const notifications = [
      {
        title: 'test 1',
        subtitle: 'subtitle test 1',
        timeStamp: '1234',
        kind: 'success'
      }, {
        title: 'test 2',
        subtitle: 'subtitle test 2',
        timeStamp: '5678',
        kind: 'success'
      }
    ]

    it('renders without crashing', () => {
      expect(shallow(<NotificationPanel notifications={notifications} />)).toMatchSnapshot()
    })

    it('renders 2 ToastNotifications', () => {
      const wrapper = mount(<NotificationPanel notifications={notifications} />)
      expect(wrapper.find('.bx--toast-notification')).toHaveLength(2)
    })
  })

  describe('when an object in the notifications array doesnt contain the kind property', () => {
    const notifications = [
      {
        title: 'test 1',
        subtitle: 'subtitle test 1',
        timeStamp: '1234'
      }
    ]

    it('defaults to the "error" type of notification', () => {
      const wrapper = mount(<NotificationPanel notifications={notifications} />)
      expect(wrapper.find('.bx--toast-notification--error')).toHaveLength(1)
    })
  })
})
