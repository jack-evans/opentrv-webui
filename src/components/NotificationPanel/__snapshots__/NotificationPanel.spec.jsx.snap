// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationPanel.jsx when notifications is a populated array renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationPanel
    notifications={
      Array [
        Object {
          "kind": "success",
          "subtitle": "subtitle test 1",
          "timeStamp": "1234",
          "title": "test 1",
        },
        Object {
          "kind": "success",
          "subtitle": "subtitle test 2",
          "timeStamp": "5678",
          "title": "test 2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ToastNotification
          caption="Time stamp: 1234"
          hideCloseButton={false}
          iconDescription="closes notification"
          kind="success"
          notificationType="toast"
          onCloseButtonClick={[Function]}
          role="alert"
          subtitle="subtitle test 1"
          title="test 1"
        />,
        <ToastNotification
          caption="Time stamp: 5678"
          hideCloseButton={false}
          iconDescription="closes notification"
          kind="success"
          notificationType="toast"
          onCloseButtonClick={[Function]}
          role="alert"
          subtitle="subtitle test 2"
          title="test 2"
        />,
      ],
      "className": "NotificationPanel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "caption": "Time stamp: 1234",
          "hideCloseButton": false,
          "iconDescription": "closes notification",
          "kind": "success",
          "notificationType": "toast",
          "onCloseButtonClick": [Function],
          "role": "alert",
          "subtitle": "subtitle test 1",
          "title": "test 1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "caption": "Time stamp: 5678",
          "hideCloseButton": false,
          "iconDescription": "closes notification",
          "kind": "success",
          "notificationType": "toast",
          "onCloseButtonClick": [Function],
          "role": "alert",
          "subtitle": "subtitle test 2",
          "title": "test 2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ToastNotification
            caption="Time stamp: 1234"
            hideCloseButton={false}
            iconDescription="closes notification"
            kind="success"
            notificationType="toast"
            onCloseButtonClick={[Function]}
            role="alert"
            subtitle="subtitle test 1"
            title="test 1"
          />,
          <ToastNotification
            caption="Time stamp: 5678"
            hideCloseButton={false}
            iconDescription="closes notification"
            kind="success"
            notificationType="toast"
            onCloseButtonClick={[Function]}
            role="alert"
            subtitle="subtitle test 2"
            title="test 2"
          />,
        ],
        "className": "NotificationPanel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "caption": "Time stamp: 1234",
            "hideCloseButton": false,
            "iconDescription": "closes notification",
            "kind": "success",
            "notificationType": "toast",
            "onCloseButtonClick": [Function],
            "role": "alert",
            "subtitle": "subtitle test 1",
            "title": "test 1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "caption": "Time stamp: 5678",
            "hideCloseButton": false,
            "iconDescription": "closes notification",
            "kind": "success",
            "notificationType": "toast",
            "onCloseButtonClick": [Function],
            "role": "alert",
            "subtitle": "subtitle test 2",
            "title": "test 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NotificationPanel.jsx when notifications is an empty array renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationPanel
    notifications={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "NotificationPanel",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "NotificationPanel",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
